flutter/lib/src/default_integrations.dart

```diff
@@ -370,15 +370,27 @@ class LoadReleaseIntegration extends Integration<SentryFlutterOptions> {
         if (name.isEmpty) {
           // Not all platforms have a packageName.
           // If no packageName is available, use the appName instead.
-          name = _cleanAppName(packageInfo.appName);
+          name = _cleanString(packageInfo.appName);
+        }
+
+        final version = _cleanString(packageInfo.version);
+        final buildNumber = _cleanString(packageInfo.buildNumber);
+
+        var release = name;
+        if (version.isNotEmpty) {
+          release = '$release@$version';
+        }
+        // At least windows sometimes does not have a buildNumber
+        if (buildNumber.isNotEmpty) {
+          release = '$release+$buildNumber';
         }
 
-        final release =
-            '$name@${packageInfo.version}+${packageInfo.buildNumber}';
         options.logger(SentryLevel.debug, 'release: $release');
 
         options.release = options.release ?? release;
-        options.dist = options.dist ?? packageInfo.buildNumber;
+        if (buildNumber.isNotEmpty) {
+          options.dist = options.dist ?? packageInfo.buildNumber;
```

Comment:
```suggestion
          options.dist = options.dist ?? buildNumber;
```

good catch, maybe use `buildNumber` which is "cleaned" already